box_connector.py

```diff
@@ -0,0 +1,245 @@
+        # The status and progress messages are more important.
+        access_token = self.box._get_auth_token()
+        if access_token is None:
+            self.save_progress("Whoops, something went wrong in retrieving a token: " + str(access_token))
+            return action_result.set_status(phantom.APP_ERROR)
+        headers = {"Authorization": "Bearer " + str(access_token)}
+
+        result = self.box._make_rest_call(url="https://api.box.com/2.0/folders/0", headers=headers)
+        if result.get("type") != "error":
+            self.save_progress("Test Connectivity Passed")
+            return action_result.set_status(phantom.APP_SUCCESS)
+        else:
+            self.save_progress("Test Connectivity Failed")
+            return action_result.set_status(phantom.APP_ERROR)
+
+    def _handle_upload_file(self, param):
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        # Implement the handler here
+        # use self.save_progress(...) to send progress messages back to the platform
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        name = param["file_name"]
+        folder_id = param["folder_id"]
+        vault_id = param["vault_id"]
+
+        try:
+            success, _, data = ph_rules.vault_info(vault_id=vault_id)
+            if not success:
+                return action_result.set_status(phantom.APP_ERROR, "Not found vault for {} id.".format(vault_id))
+            data = list(data)[0]
+            file_path = data.get('path')
+            file = open(file_path, "rb")
+        except Exception as e:
+            return action_result.set_status(phantom.APP_ERROR, "There was an issue opening the file: " + str(e.message))
+
+        result = self.box._create_file(name=name, parent_id=folder_id, file=file)
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+
+        action_result.add_data(result)
+        summary = {}
+        if result.get('type') != 'error' and result.get('entries'):
+            entries = result.get('entries')[0]
+            summary['file_id'] = str(entries.get('id'))
+            summary['sha1'] = entries.get('sha1')
+            summary = action_result.update_summary(summary)
+        else:
+            return action_result.set_status(phantom.APP_ERROR, "There was an issue with the file creation: " + str(result))
+        # Add a dictionary that is made up of the most important values from data into the summary
+        # BaseConnector will create a textual message based off of the summary dictionary
+        return action_result.set_status(phantom.APP_SUCCESS, "Successfully uploaded file")
+
+    def _handle_create_folder(self, param):
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        # Implement the handler here
+        # use self.save_progress(...) to send progress messages back to the platform
```

Comment:
Seeing that we're inside the handler function, do you think the comments here sound redundant?

Reply:
Yes, Deleted redundant comments.